File: Sorting/JavaScript/bubbleSort.js
function bubbleSort(arr) {
    let swapped;
    for (let i = arr.length; i > 0; i--) {
        swapped = false;
        for (let j = 0; j < i - 1; j++) {
            console.log(arr, arr[j], arr[j + 1]);
            if (arr[j] > arr[j + 1]) {
                let temp = arr[j];
                arr[j] = arr[j + 1];
                arr[j + 1] = temp;
                swapped = true;
            }
        }
        console.log("A pass has completed");
        if (!swapped) break;
    }

    return arr;
}

bubbleSort([8, 6, 7, 4, 5, 3, 1, 2]);

File: Sorting/Python/bubble_sort.py
def bubble_sort(arr):
    for i in range(len(arr) - 1, -1, -1):
        swapped = False
        for j in range(0, i):
            print(arr, arr[j], arr[j + 1])
            if arr[j] > arr[j + 1]:
                arr[j], arr[j + 1] = arr[j + 1], arr[j]
                swapped = True
        print('made a pass')
        if not swapped:
            break
    return arr


def bubble_sort2(arr):
    swapped = True

    while swapped:
        swapped = False
        for i in range(len(arr) - 1):
            if arr[i] > arr[i + 1]:
                arr[i], arr[i + 1] = arr[i + 1], arr[i]
                swapped = True

    return arr
